src/Transform/ONNX/ConstProp.cpp

```diff
@@ -1008,6 +1008,19 @@ Value ConstPropRange(PatternRewriter &rewriter, Value replacingValue,
   return createReplacingConstantOp(rewriter, replacingValue, rangeElements);
 }
 
+//===----------------------------------------------------------------------===//
+// Code to perform constant propagation for NonZero.
+//===----------------------------------------------------------------------===//
+
+Value ConstPropNonZero(
+    PatternRewriter &rewriter, Value replacingValue, Value constValue) {
+  ConstPropCounters::count("NonZero", {constValue});
+  ElementsAttr constElements = getConstValueElements(constValue);
+  OnnxElementsAttrBuilder elementsBuilder(rewriter.getContext());
+  ElementsAttr nonZeroElements = elementsBuilder.nonZero(constElements);
+  return createReplacingConstantOp(rewriter, replacingValue, nonZeroElements);
```

Comment:
@sorenlassen @chentong319 to make progress, could we do function type update inside `createReplacingConstantOp`? I know it's not the final solution, just would like to have NonZero constant prop landed since we see it in GPT-2 models.

Reply:
we could go ahead and introduce the new onnx function return op, that we've discussed, see PR #2239